platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/container/ContainerTransactionGenerator.java

```diff
@@ -13,19 +23,91 @@ public class ContainerTransactionGenerator implements TransactionGenerator {
 
     private static final Logger log = LogManager.getLogger();
 
+    /** Duration between two transaction submissions to a single node. */
+    private static final Duration CYCLE_DURATION = Duration.ofSeconds(1).dividedBy(TPS);
+
+    /** Secure random instance used to create pseudo-random payloads. */
+    private final SecureRandom random = new SecureRandom();
+
+    /** Supplies the current list of nodes to which transactions should be sent. */
+    private Supplier<List<Node>> nodesSupplier = List::of;
+
+    private ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+
+    private volatile boolean running = false;
+
+    /**
+     * Sets the supplier that provides the up-to-date list of nodes. Must be called by the network once
+     * the nodes are available.
+     *
+     * @param supplier supplier returning the current nodes
+     */
+    public void setNodesSupplier(final Supplier<List<Node>> supplier) {
+        this.nodesSupplier = supplier;
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     public void start() {
-        log.warn("Starting ContainerTransactionGenerator not implemented yet.");
+        if (running) {
+            return;
+        }
+
+        /*
+         * If the generator was stopped previously, the scheduler may have been shut down. In that case
+         * create a fresh scheduler instance so the generator can be restarted safely.
+         */
+        if (scheduler.isShutdown()) {
+            scheduler = Executors.newSingleThreadScheduledExecutor();
```

Comment:
I think we should not shut down the `scheduler`, but cancel the `Future` that is returned when scheduling the work. It may also be worth considering to inject the `scheduler`, making unit testing easier.